wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/MemoryUtil.java

```diff
@@ -0,0 +1,73 @@
+package com.github.bordertech.wcomponents.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * This class is intended to help detect and trace possible memory leaks.
+ * The goals are:
+ * - "Make some noise" when lists of UI components continue to grow beyond a reasonable level.
+ * - Use a common logger so that these messages can be easily silenced for all classes.
+ * - Provide a stack trace so that possible memory leaks are easier to track.
+ *
+ * @author Rick Brown
+ */
+public final class MemoryUtil {
+
+	/**
+	 * For convenience this can be used as a "rule of thumb" when testing the size of a List of components in the UI.
+	 */
```

Comment:
What if this was a runtime parameter that defaults to 1000. This would allow projects to adjust the value.